unit-tests/unit-tests-live.cpp

```diff
@@ -5915,3 +5915,114 @@ TEST_CASE("l500_presets_set_preset", "[live]")
+                        auto vid = sp.as<rs2::video_stream_profile>();
+                        auto h = vid.height();
+                        auto w = vid.width();
+                        if (!(w == profile.width && h == profile.height)) continue;
+                        if (sp.stream_type() == RS2_STREAM_COLOR)
+                        {
+                            rgb_stream_profiles.push_back(sp);
+                            rgb_sensor = s;
+                        }
+                    }
+                }
+            }
+            auto process_frame = [&](const rs2::frame& f)
+            {
+                std::lock_guard<std::mutex> lock(mutex_process);
+                auto stream_type = f.get_profile().stream_name();
+                auto frame_num = f.get_frame_number();
+                auto unique_id = f.get_profile().unique_id();
+                if (std::find(rgb_frames_num.begin(), rgb_frames_num.end(), frame_num) != rgb_frames_num.end()) // check if frame is already processed
+                    return;
+                rgb_frames_num.push_back(frame_num);
+            };
+            auto frame_callback = [&](const rs2::frame& f)
+            {
+                std::lock_guard<std::mutex> lock(mutex);
+                if (rs2::frameset fs = f.as<rs2::frameset>())
+                {
+                    // With callbacks, all synchronized stream will arrive in a single frameset
+                    for (const rs2::frame& ff : fs)
+                    {
+                        process_frame(ff);
+                    }
+                }
+                else
+                {
+                    // Stream that bypass synchronization (such as IMU) will produce single frames
+                    process_frame(f);
+                }
+            };
+            
+            for (auto i = 0; i < 3; i++)
+            {
+                std::cout << "==================================" << std::endl;
+                rgb_frames_num.clear();
+                rgb_sensor.open(rgb_stream_profiles);
+                rgb_sensor.start(frame_callback);
+                int count_drops = 0; //frame drops counter will increase if 2 or more successive frames are missing
+                unsigned long long prev_frame_num = 0;
+                std::this_thread::sleep_for(std::chrono::seconds(30));
+                std::lock_guard<std::mutex> lock(mutex);
+                // analysis : check of >2 consecutive frames are missing
+                for (auto f : rgb_frames_num)
+                {
+                    if (prev_frame_num > 0)
+                    {
+                        if (f - prev_frame_num > 2)
+                            count_drops += 1;
+                    }
+                    prev_frame_num = f;
+                }
+                std::cout << "Iteration " << i << " : frame drops = " << count_drops << std::endl;
```

Comment:
You 're performing the calculation while the sensor is still streaming, why?
1.Simplify the code and the remove mutex in line 6007.
2.The test shall include at least one `REQUIRE(...)` for actual results in that iteration.
3. Make it run either for at least for 30 min before stop